### PR TITLE
chore: organize demos

### PR DESCRIPTION
**Pull request by xavierchanth**
**- What I did**

- Moved all demos into demos/
- Moved at_demo_data into packages/

**- How I did it**

**- How to verify it**

**- Description for the changelog**
chore: organize demos

### CODE REVIEW

**Comment by cpswan** [comment]
@XavierChanth is this ready to review now?

**Comment by xavierchanth** [comment]
> @XavierChanth is this ready to review now?

Yes it is

File: .github/workflows/static_analysis.yml [hunk 1]
```diff
@@ -24,41 +24,41 @@ jobs:
...
       - name: Install dependencies and run analyze - at_data_share
-        working-directory: ./at_data_share/
+        working-directory: ./demos/at_data_share/
         run: |
           dart pub get
           dart analyze
 
       # Runs osv-scanner to find any vulnerable Dart dependencies
-      # It needs to look at pubspec.lock files, which is why it's
+      # It needs to look at pubspec.lock ranls, which is why it's
```
Comment (cpswan) [inline .github/workflows/static_analysis.yml]: What's a `ranls`?

**Review by cpswan — CHANGES_REQUESTED** [review]
Mostly LGTM, but one change that looks accidental, and also `mwc_demo/dart/iot_sender/bin/.set_hro2_receivers.dart.un~` shouldn't be there, so let's use this opportunity to get rid.

File: .github/workflows/static_analysis.yml (hunk 1 above)
Comment (xavierchanth) [reply .github/workflows/static_analysis.yml]: accidental change...

**Review by cpswan — APPROVED** [review]

**Comment by cpswan** [comment]
@XavierChanth can we also lose the stray vim undo file at `mwc_demo/dart/iot_sender/bin/.set_hro2_receivers.dart.un~`

**Comment by xavierchanth** [comment]
> @XavierChanth can we also lose the stray vim undo file at `mwc_demo/dart/iot_sender/bin/.set_hro2_receivers.dart.un~`

That was a delete

**Pull request merged**